controller_manager:
  ros__parameters:
    # update_rate: 10  # Hz
    update_rate: 5  # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    diff_drive_controller:
      type: diff_drive_controller/DiffDriveController

    joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

diff_drive_controller:
  ros__parameters:
    publish_rate: 10.0
    left_wheel_names: ['base_to_front_left_leg']
    right_wheel_names: ['base_to_front_right_leg']
    wheels_per_side: 1
    wheel_separation: 0.25
    wheel_radius: 0.04
    base_frame_id: base_link
    use_stamped_vel: false
    pose_covariance_diagonal: [0.0001, 0.0001, 0.0001, 0.0001, 0.0001, 0.001]
    twist_covariance_diagonal: [0.0001, 0.0001, 0.0001, 0.0001, 0.0001, 0.001]
    # pose_covariance_diagonal: [0.0001, 0.0001, 0.0001, 0.01, 0.01, 0.001]
    # twist_covariance_diagonal: [0.0001, 0.0001, 0.0001, 0.01, 0.01, 0.001]

    linear.x.has_velocity_limits: false
    linear.x.has_acceleration_limits: false
    linear.x.has_jerk_limits: false
    linear.x.max_velocity: 0.0
    linear.x.min_velocity: 0.0
    linear.x.max_acceleration: 0.0
    linear.x.min_acceleration: 0.0
    linear.x.max_jerk: 0.0
    linear.x.min_jerk: 0.0

    angular.z.has_velocity_limits: false
    angular.z.has_acceleration_limits: false
    angular.z.has_jerk_limits: false
    angular.z.max_velocity: 0.0
    angular.z.min_velocity: 0.0
    angular.z.max_acceleration: 0.0
    angular.z.min_acceleration: 0.0
    angular.z.max_jerk: 0.0
    angular.z.min_jerk: 0.0

joint_trajectory_controller:
  ros__parameters:
    joints:
      - shoulder_to_shoulder_joint
      - upper_arm_to_elbow

    command_interfaces:
      - velocity
     
    state_interfaces:
      - position
      - velocity

    gains: 
      shoulder_to_shoulder_joint: {p: 2.0, i: 5.0}
      upper_arm_to_elbow: {p: 2.0, i: 5.0}
